pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -1226,6 +1226,40 @@ func (c *Controller) WorkloadInstanceHandler(si *model.WorkloadInstance, event m
 		c.workloadInstancesIndex.Insert(si)
 	}
 
+	// handle WorkloadInstance in ambient by creating a mock pod that models the VM workload.
+	ambientMockPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      si.Name,
+			Namespace: si.Namespace,
+			Labels:    si.Endpoint.Labels,
+		},
+		Spec: v1.PodSpec{
+			ServiceAccountName: si.Endpoint.ServiceAccount,
+			NodeName:           si.Endpoint.NodeName,
+			Containers: []v1.Container{
+				{
+					Ports: []v1.ContainerPort{},
+				},
+			},
+		},
+		Status: v1.PodStatus{
+			PodIP: si.Endpoint.Address,
+			Conditions: []v1.PodCondition{{
+				Type:   v1.PodReady,
+				Status: v1.ConditionTrue,
+			}},
+		},
+	}
+	// treat workloads on ambient as HBONE native
+	ambientMockPod.Labels[model.TunnelLabel] = model.TunnelHTTP
```

Comment:
I don't think this should be intrinsic. The user should stick this on their workload entry if it's running in that mode.

Reply:
okay, will delete. Just going to let it be passed through si.Endpoint.Labels